### PR TITLE
Fix crash when launching the app on MacOS

### LINKED ISSUE #346: Run on MacOS with Apple Silicon
Some (most?) iOS apps can be run on MacOS with Apple Silicon: https://support.apple.com/en-il/guide/app-store/fird2c7092da/mac
For example, I can run the official Israel Railway iOS app on my mac.
However, the Better Rail app is not available on the app store for mac (even not as an iOS app).

### LINKED ISSUE #348: Active ride background color on search screen is identical to short ride background
image

### PR DESCRIPTION
**Pull request by guytepper**
The crash occurred because the live activities monitoring method initialized on the Mac app, where it's not available

Resolves #346

### CODE REVIEW

File: app/utils/ios-helpers.tsx [hunk 1]
```diff
@@ -3,7 +3,11 @@ import { RouteItem } from "../services/api"
 
 const { RNBetterRail } = NativeModules
 
-export const canRunLiveActivities = Platform.OS == "ios" && parseFloat(Platform.Version) >= 16.2
+function isRunningOnMac() {
+  return RNBetterRail.isRunningOnMac()
```
Comment (planecore) [inline app/utils/ios-helpers.tsx:7]: Isn't this a promise?

**Review by planecore — CHANGES_REQUESTED** [review]

File: app/utils/ios-helpers.tsx [hunk 2]
```diff
@@ -3,7 +3,16 @@ import { RouteItem } from "../services/api"
...
-export const canRunLiveActivities = Platform.OS == "ios" && parseFloat(Platform.Version) >= 16.2
+export async function canRunLiveActivities() {
+  const isRunningOnMac = await RNBetterRail.isRunningOnMac()
+  const deviceSupported = Platform.OS == "ios" && parseFloat(Platform.Version) >= 16.2
+
+  if (deviceSupported && !isRunningOnMac) {
+    return true
+  }
+
+  return false
```
Comment (planecore) [inline app/utils/ios-helpers.tsx:14]:
```suggestion
  return deviceSupported && !isRunningOnMac
```

File: app/app.tsx [hunk 3]
```diff
@@ -70,10 +70,12 @@ function App() {
 
   useEffect(() => {
     // Activate live activities listener on iOS 16.2+
-    if (canRunLiveActivities) {
-      monitorLiveActivities()
-    }
-  }, [])
+    rootStore?.ride.checkLiveActivitiesSupported().then((canRunLiveActivities) => {
+      if (canRunLiveActivities === true) {
```
Comment (planecore) [inline app/app.tsx:74]:
```suggestion
      if (canRunLiveActivities) {
```

**Review by planecore — CHANGES_REQUESTED** [review]

File: app/app.tsx (hunk 3 above)
Comment (guytepper) [reply app/app.tsx:74]: I like it better when it's explicit, more readable and the intention is clearer

Comment (planecore) [reply app/app.tsx:74]: It's not a conversion we currently have in the app. We don't have this anywhere in our source - but it's your call. If you like it feel free to resolve this.

**Review by planecore — APPROVED** [review]

**Pull request merged**